src/test/compile-fail/bad-mid-path-type-params.rs

```diff
@@ -39,12 +39,14 @@ impl Trait<isize> for S2 {
 fn foo<'a>() {
     let _ = S::new::<isize,f64>(1, 1.0);
     //~^ ERROR too many type parameters provided
+    //~| ERROR mismatched types
```

Comment:
This error occurs because the new code doesn't use `TypeErr` but instead uses the first N parameters and then errors on the rest -- seems like this has a reasonable chance of being spurious, I think I would prefer to preserve the old behavior.